cmd/web/main.go

```diff
@@ -65,43 +65,31 @@ func main() {
 	})
 	flag.Parse()
 
-	infoLog := log.New(os.Stdout, "INFO ", log.Ldate|log.Lshortfile)
-	errorLog := log.New(os.Stderr, "ERROR ", log.Ldate|log.Lshortfile)
+	zerolog.TimeFieldFormat = time.RFC3339
 
 	db, err := openDB(cfg)
 	if err != nil {
-		errorLog.Fatalln(err)
+		log.Fatal().Err(err).Send()
 	}
-	infoLog.Println("Coonected to DB successfully")
+	log.Info().Msg("Coonected to DB successfully")
 
-	driver, err := postgres.WithInstance(db, &postgres.Config{})
-	if err != nil {
-		errorLog.Fatalln(err)
-	}
-	m, err := migrate.NewWithDatabaseInstance(
-		"file://migrations",
-		"postgres", driver)
+	err = autoMigrate(db)
 	if err != nil {
-		errorLog.Fatalln(err)
-	}
-	err = m.Up()
-	if err != nil && !errors.Is(err, migrate.ErrNoChange) {
-		errorLog.Fatalln(err)
+		log.Fatal().Err(err).Send()
 	}
 
-	emailModel := &models.EmailModel{DB: db}
+	emailRepository := &repositories.PostgresEmailRepository{DB: db}
+	emailService := services.NewEmailService(emailRepository)
 	rateService := services.NewRateService(time.Hour)
 	rateService.StartBackgroundTask()
 
-	mailerService := services.NewMailerService(cfg.mailer, emailModel, rateService, errorLog)
+	mailerService := services.NewMailerService(cfg.mailer, emailService, rateService)
```

Comment:
Taking into account OCP, consider making this service more abstract to have ability in future extend with other ways to notify clients about exchange rate. E.g. there could be SMS or push notifications.

Reply:
You're right! I'll handle it